--- ios.pm.orig	2024-12-28 14:34:51.177321726 +0700
+++ ios.pm	2024-12-28 14:42:00.026629029 +0700
@@ -2372,6 +2372,10 @@
 	    ProcessHistory("","","", "!\nconfig-register $config_register\n");
 	}
 
+	# Skip lines containing timestamp_write
+	next if (/^\s*timestamp_write:/);
+	next if (/[+-]\s*timestamp_write:/);
+
 	/Non-Volatile memory is in use/ && return(-1); # NvRAM is locked
 	/% Configuration buffer full, / && return(-1); # buffer is in use
 	$linecnt++;
@@ -2796,3 +2800,4 @@
 }
 
 1;
+
